### DISCUSSION TITLE
checkboxlist

### DISCUSSION THREAD

**Discussion by photojo** [filamentphp/filament#4450] [Help]
Hallo!
I'm a beginner in laravel and filament.
I've implemented a filament form with a checkboxlist, described here: https://filamentphp.com/docs/2.x/forms/fields#checkbox-list 
My Checkboxlist:: renders correctly 5 kind of fruits

[ ] **apple**
[ ] orange
[ ] pear
[ ] **banana**
[ ] **kiwi**

My values are saved in the DB column as ['**apple**','**banana**','**kiwi**'] when those are checked.
Do you have a more concrete example or tutorial, how to get the array FROM the DB and set the boxes to checked?
Tanks a lot.
Jo

**Comment by danharrin** [comment]
You just need to make sure that the column has an `array` cast in your model

**Comment by photojo** [comment]
model

```
class Fruits extends Model
{
    use HasFactory;
    protected $table = 'myfruits';
    protected $casts = [];
}
```

component 
    
 ```
protected function getFormSchema(): array
    {
        return [
            CheckboxList::make('casts')
                ->label('')
                ->options([
                    'apple',
                    'orange',
                    'pear',
                    'banana',
                    'kiwi'
                ])
        ];
    }

```

**Reply by danharrin** [reply]
```php
class Fruits extends Model
{
    use HasFactory;
    protected $table = 'myfruits';
    protected $casts = [
        'fruit' => 'array',
    ];
}
```

```php
public function mount()
{
    $fruits = Fruits::first();

    $this->form->fill([
        'fruit' => $fruits?->fruit ?? [],
    ]);
}

protected function getFormSchema(): array
{
   return [
       CheckboxList::make('fruit')
           ->label('Fruit')
           ->options([
               'apple',
               'orange',
               'pear',
               'banana',
               'kiwi'
           ])
   ];
}
```

**Reply by photojo** [reply]
Thank you very very much! It works fine! You saved my day!

Greetings
Jo

**Reply by photojo** [reply]
I'm so sorry, it sounds stupid, but I have an other simply problem with this item:
In my CheckboxList I added 
->reactive()
->afterStateUpdated(fn($state) => $this->submit())

How can I send my fruit-Array to the submit-Routine? ...submit( ??? ) I tried several things without success.

Thank you!

**Reply by danharrin** [reply]
well you could pass it in `$this->submit($state)` or call `$this->form->getState()` from inside `submit()`

**Comment by photojo** [comment]
Ok, I already tried this. And to see if it works, I've made ...

```
  public function submit(): void
    { 
    $var = $this->form->getState();
        Notification::make()
            ->title('Notification')
            ->body($var)
     ....
    }
```

to see the result. Obviously this doesn't work?! Finally I have to execute for each element in the fruits-array a DB insert command.

P.S.: I'm always looking for a possibility like 'console.log(...)' in Angular. So I tried so "log" the array in the notification.

**Comment by photojo** [comment]
Please allow me a last question to this topic. How can I get just the fruits-array out of "$this->form->getState();" in my submit-function to call another function for each element in this array.
Thank you so much in advance.
